File: 0002/solution.cpp
#include <iostream>

#define INPUT 4000000

using namespace std;

long int fibonacci(const int n)
{
	if(n == 0 || n == 1) {
		return n;
	}

	else return fibonacci(n-1) + fibonacci(n-2);
}

int solve(const int threshold) {
    long int solution {0}, auxiliar {0};
    int i {1};

    auxiliar = fibonacci(i);
    while(auxiliar < threshold) {
        if(auxiliar % 2 == 0)
            solution += auxiliar;
        i++;
        auxiliar = fibonacci(i);
    }

    return solution;
}

int main(void) {
	cout << solve((const int) INPUT) << endl;
	return 0;
}

File: 0002/solution.py
#!/usr/bin/python3

from functools import cache

@cache
def fib(bar: int) -> int:
    if bar < 2:
        return bar
    return fib(bar-1) + fib(bar-2)

def solve(threshold: int) -> (int):
    solution, auxiliar = 0, 0
    i = 1

    auxiliar = fib(i)
    while auxiliar < threshold:
        if (auxiliar % 2 == 0):
            solution += auxiliar
        i=i+1
        auxiliar = fib(i)

    return solution

if __name__ == '__main__':
    print(solve(4000000))
